### ISSUE TITLE
When syncing with YT Music only 100 tracks are loaded into the playlist

### ISSUE THREAD

**Issue by MAX-TAC** [toasterofbread/spmp#305]
### Checklist

- [ ] I am able to reproduce the bug with the latest debug version ([Android](https://nightly.link/toasterofbread/spmp/workflows/build-android/main), [Linux](https://nightly.link/toasterofbread/spmp/workflows/build-linux/main), [Windows](https://nightly.link/toasterofbread/spmp/workflows/build-windows/main))
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce

If you go to the playlist Favorite music that is synchronized YT Music then it displays only 100 tracks. It’s also annoying that you have to load tracks when you try to scroll to the very end of the playlist.
![Screenshot_2024-05-13-15-57-10-324_com toasterofbread spmp-edit](https://github.com/toasterofbread/spmp/assets/115115023/b98f4baa-ab7c-4f3d-b73c-ac7275d2ab3e)
![Screenshot_2024-05-13-15-57-36-266_com toasterofbread spmp-edit](https://github.com/toasterofbread/spmp/assets/115115023/6b3ba088-a8d3-4164-bb63-daea9d4a6d22)

### Expected behavior

Please fix the bug to download the full volume of the playlist and, if possible, remove the download button for tracks at the bottom of the list when scrolling to the end of the playlist.

### Actual behavior

-

### Logs

-

### SpMp version

3.2

### SpMp platform

Android

### OS version

Android 14 Hyper OS 1.0.2.0

**Comment by toasterofbread** [comment]
I'm not entirely sure what the issue is. Does the load button not work?

**Comment by MAX-TAC** [comment]
No, the problem is that in my playlist YouTube Music 312 track, and in SpMp Only 100 tracks are displayed and loaded.
![Screenshot_2024-05-13-16-10-54-042_com toasterofbread spmp-edit](https://github.com/toasterofbread/spmp/assets/115115023/252dacf7-07c6-479c-9758-2f3181c305c0)
![Screenshot_2024-05-13-16-11-13-308_com google android apps youtube music-edit](https://github.com/toasterofbread/spmp/assets/115115023/1d652715-f475-4eb7-866f-40aa20c989c7)

**Comment by toasterofbread** [comment]
Also, this is a duplicate of #261 which should have been fixed by https://github.com/toasterofbread/spmp/commit/2269bd0e73c4400fc455795407d7bb1de8877a66.

**Comment by MAX-TAC** [comment]
Regarding the download button when rewinding a playlist to the end, this is most likely another bug. When I scroll through the playlist to the very bottom there is a button to load the remaining tracks, when I press it,the tracks appear, but if you go back to the very top, there will be tracks not from the beginning of the playlist, but from approximately the middle. In order for tracks from the beginning of the playlist to appear, you have to update the playlist swipe to update the playlist page.

**Comment by toasterofbread** [comment]
Which commit does this occur on?

**Comment by MAX-TAC** [comment]
> Да, для этого и нужна кнопка загрузки внизу плейлиста (она продолжает плейлист). Это не работает?
This button doesn't work. When loading, the playlist displays approximately 50 tracks, this button is like scrolling through the playlist and shows not 100 tracks but only the remaining 50. Also, this button does not synchronizes playlist YT Music.

**Comment by MAX-TAC** [comment]
> Кроме того, это дубликат #261 , который должен был быть исправлен в [2269bd0](https://github.com/toasterofbread/spmp/commit/2269bd0e73c4400fc455795407d7bb1de8877a66) .

Apparently the bug hasn't gone away YpMp Synchronizes only 100 tracks from the playlist of your favorite music YT Music.

**Comment by toasterofbread** [comment]
Again, which commit does this occur on? (You marked `I am able to reproduce the bug with the latest debug version`)

**Comment by MAX-TAC** [comment]
> На каком коммите это происходит?

SpMp release build v0.3.2

**Comment by toasterofbread** [comment]
You ticked the box saying you're able to reproduce this with the latest *debug* version. Which version did you use?

**Comment by MAX-TAC** [comment]
Sorry, I accidentally checked this box. I’ll be honest, I’m an ordinary user who understands a little about development and testing, so I can reproduce this bug in the debug I can't do the version.

**Comment by toasterofbread** [comment]
That's OK. If you do end up testing a debug build (linked next to the check box) and find the bug still occurs, please report at #261.

**Comment by MAX-TAC** [comment]
OK, thank you. I have already installed the debug version, the tracks are loaded in it, but the track counter in the playlist does not work adequately. Each time you scroll through the playlist to the very end, it adds up the number of tracks each time, and when you update the playlist, it resets to 100 as it was before. I need to write a new bug report or write this in #261?

**Comment by MAX-TAC** [comment]
![Screenshot_2024-05-13-16-41-47-593_com toasterofbread spmp debug](https://github.com/toasterofbread/spmp/assets/115115023/1d730965-f1ac-4a64-a5f4-3d31449178c6)
![Screenshot_2024-05-13-16-42-06-803_com toasterofbread spmp debug](https://github.com/toasterofbread/spmp/assets/115115023/d5e084b1-e9bb-4dab-bdbc-9f1cf511cbdd)
![Screenshot_2024-05-13-16-42-49-771_com toasterofbread spmp debug](https://github.com/toasterofbread/spmp/assets/115115023/5b82cd7c-71e3-4748-b3ad-e5560fe5aed6)
Pay attention to the number of tracks in the playlist. So, every time you scroll to the end of the playlist, the number of tracks is summed up.

**Comment by toasterofbread** [comment]
> Each time you scroll through the playlist to the very end, it adds up the number of tracks each time, and when you update the playlist, it resets to 100 as it was before.

Unless the counter is displaying an incorrect sum (for the amount of currently loaded items), this is intended behaviour.

**Comment by MAX-TAC** [comment]
Tell me, was the fix for this bug #261 released in version 3.2? Or when can we expect a release version with a fix?

**Comment by toasterofbread** [comment]
The fix wasn't in v0.3.2. I can't give an estimate for the next release right now, but it should be about a week or two after #228 is merged.
